### PR TITLE
Feature/first quest

### PR DESCRIPTION
**Pull request by tikisailor**
In order to run we need to add three collections to the degen db: `firstQuestContent`, `firstQuestContentBackup` and `firstQuestTracker`. Also `firstQuestContent` needs to be pre-seeded with a document to hold the current message content of first-quest. Please let me know how we can go about it. 

With regards to roles: I used the existing roles currently used in first-quest as means to be able to continue the quest easily where users left off should the collector time out before the user moves to the next step. I also added two new roles (`unverified` and `verified`). We would have to create those and maybe discuss if the naming is ok. `unverified` is the role users get assigned when first entering the server. Ideally channels should be configured to minimal/no access for unverified role. After clicking the verification reaction it gets removed and the user becomes `verified`. Throughout the first quest with every step the user completes they get assigned a new role and the old role gets removed. At the end of the quest they remain with `First Quest Complete` role.

### CODE REVIEW

File: src/app/service/constants/constants.ts [hunk 1]
```diff
@@ -38,4 +41,16 @@ export default Object.freeze({
 	
 	POAP_REQUIRED_PARTICIPATION_DURATION: Number(process.env.POAP_REQUIRED_PARTICIPATION_DURATION),
 	POAP_MAX_DURATION_MINUTES: Number(process.env.POAP_MAX_EVENT_DURATION_MINUTES),
+
+	FIRST_QUEST_ROLES : {
+		unverified: 'unverified',
```
[the remark this answers is not in the document]
Comment (tikisailor) [reply src/app/service/constants/constants.ts:46]: yes, will be removed. Will submit new implementation along with captcha verification

[the remark this answers is not in the document]
Comment (tikisailor) [reply src/app/service/constants/constants.ts]: done

File: src/app/events/GuildMemberAdd.ts [hunk 2]
```diff
@@ -12,6 +13,20 @@ export default class implements DiscordEvent {
 			if (ServiceUtils.isBanklessDAO(member.guild)) {
 				if (await ServiceUtils.runUsernameSpamFilter(member)) {
 					return;
+				} else {
+					if (!(member.user.bot)) {
```
[the remark this answers is not in the document]
Comment (tikisailor) [reply src/app/events/GuildMemberAdd.ts]: done

**Comment by amaredeus** [comment]
As per dev standup 2 discussion, awaiting feedback from angyst, frog, and behold before merging!

**Comment by tikisailor** [comment]
> As per dev standup 2 discussion, awaiting feedback from angyst, frog, and behold before merging!

Demos are done, requested changes discussed during demo are included.

File: .env.prod [hunk 3]
```diff
@@ -21,13 +21,25 @@ DISCORD_ROLE_GRANTS_COMITTEE=860222982416498688
 DISCORD_ROLE_GUEST_PASS=848369657281773599
 DISCORD_ROLE_DEVELOPERS_GUILD=843924063591727195
 
+DISCORD_ROLE_FIRST_QUEST_START=
```
Comment (amaredeus) [inline .env.prod:24]: Has this role been created?

**Review by amaredeus — CHANGES_REQUESTED** [review]
I'm seeing this message under config

```
Overview of current message content
Message 1️⃣
Hello! Welcome to BanklessDAO. We're glad you're here 🙂

We would like to encourage all new members to activate 2 factor authentication (2FA), See how it's done here: <URL REMOVED>

**The easies...
Message 2️⃣
>> Important message: Support will never send you a Direct Message offering DAO assistance. You must request it by your DM. <<
>> Never ever ever reveal your private keys.<<

-----------------...
Message 3️⃣
Welcome! The purpose of this quest is to choose your membership path.

-----------------------------------------

Participating in the DAO has no financial requirements.

However, obtaining Level 1 me...
Message 4️⃣
Welcome! The purpose of this quest is to ease into our primary communication channel, Discord.

If you’ve never used Discord before or are unfamiliar with its layout, please watch this short video:
<U...
Message 5️⃣
Welcome! The purpose of this quest is to introduce you to why BanklessDAO exists.

-----------------------------------------

Bankless is a movement for pioneers seeking liberation from the tyranny o...
Message 6️⃣
Welcome! The purpose of this quest is to provide a pathway for members without 35,000 $BANK to get involved using our Guest Pass.

-----------------------------------------

We do not want the 35,00...
Message 7️⃣
Welcome! You’ve made it to your true first quest, should you choose to accept (you should!)

-----------------------------------------

Even though you’ve made it to the DAO and have familiarized ...
Message 8️⃣
Congratulations! You have reached the end of first quest.

If you want to repeat first quest, you can reset your role and start over by responding with !first-quest.
...
select emote to edit corresponding question
```

File: .env.prod (hunk 3 above)
Comment (tikisailor) [reply .env.prod:24]: Not yet, who do i have to talk to to set it up?

**Comment by tikisailor** [comment]
> I'm seeing this message under config
> 
> ```
> Overview of current message content
> Message 1️⃣
> Hello! Welcome to BanklessDAO. We're glad you're here 🙂
> 
> We would like to encourage all new members to activate 2 factor authentication (2FA), See how it's done here: <URL REMOVED>
> 
> **The easies...
> Message 2️⃣
> >> Important message: Support will never send you a Direct Message offering DAO assistance. You must request it by your DM. <<
> >> Never ever ever reveal your private keys.<<
> 
> -----------------...
> Message 3️⃣
> Welcome! The purpose of this quest is to choose your membership path.
> 
> -----------------------------------------
> 
> Participating in the DAO has no financial requirements.
> 
> However, obtaining Level 1 me...
> Message 4️⃣
> Welcome! The purpose of this quest is to ease into our primary communication channel, Discord.
> 
> If you’ve never used Discord before or are unfamiliar with its layout, please watch this short video:
> <U...
> Message 5️⃣
> Welcome! The purpose of this quest is to introduce you to why BanklessDAO exists.
> 
> -----------------------------------------
> 
> Bankless is a movement for pioneers seeking liberation from the tyranny o...
> Message 6️⃣
> Welcome! The purpose of this quest is to provide a pathway for members without 35,000 $BANK to get involved using our Guest Pass.
> 
> -----------------------------------------
> 
> We do not want the 35,00...
> Message 7️⃣
> Welcome! You’ve made it to your true first quest, should you choose to accept (you should!)
> 
> -----------------------------------------
> 
> Even though you’ve made it to the DAO and have familiarized ...
> Message 8️⃣
> Congratulations! You have reached the end of first quest.
> 
> If you want to repeat first quest, you can reset your role and start over by responding with !first-quest.
> ...
> select emote to edit corresponding question
> ```

Yep, that is the overview of current messages to select from to update messages

File: .env.prod (hunk 3 above)
Comment (tikisailor) [reply .env.prod]: created now

Comment (tikisailor) [reply .env.prod]: created now

**Comment by amaredeus** [comment]
It's looking very good! One small update, let's include the changes in the `CHANGELOG.md` file and then we're ready to go!

**Review by amaredeus — APPROVED** [review]

**Review by amaredeus — APPROVED** [review]

**Review by nonsensecodes — APPROVED** [review]
Well done, @tikisailor! Thank you for bringing this to fruition.

**Pull request merged**